src/borg/testsuite/repo_new.py

```diff
@@ -0,0 +1,257 @@
+import os
+
+import pytest
+
+from ..helpers import IntegrityError
+from ..repository import Repository, MAX_DATA_SIZE, TAG_DELETE, TAG_PUT2, TAG_PUT, TAG_COMMIT
+from ..repoobj import RepoObj
+from .hashindex import H
+
+
+UNSPECIFIED = object()  # for default values where we can't use None
+
+
+def fchunk(data, meta=b""):
+    # create a raw chunk that has valid RepoObj layout, but does not use encryption or compression.
+    meta_len = RepoObj.meta_len_hdr.pack(len(meta))
+    assert isinstance(data, bytes)
+    chunk = meta_len + meta + data
+    return chunk
+
+
+def pchunk(chunk):
+    # parse data and meta from a raw chunk made by fchunk
+    meta_len_size = RepoObj.meta_len_hdr.size
+    meta_len = chunk[:meta_len_size]
+    meta_len = RepoObj.meta_len_hdr.unpack(meta_len)[0]
+    meta = chunk[meta_len_size : meta_len_size + meta_len]
+    data = chunk[meta_len_size + meta_len :]
+    return data, meta
+
+
+def pdchunk(chunk):
+    # parse only data from a raw chunk made by fchunk
+    return pchunk(chunk)[0]
+
+
+class RepositoryTestCaseBase:
+    def reopen(self, repository, exclusive=UNSPECIFIED):
+        repository.__exit__(None, None, None)
+        repository.__enter__()
+        repository.exclusive = exclusive
```

Comment:
this could maybe become a `Repository` method.

Reply:
btw, one reason why this was done like that is that the unittest-style base class had a setup/teardown method that entered/exited the repository context. So the exit/enter in `reopen` was to work within that given setup/teardown environment.

Reply:
With this in mind I tried to replicate that behavior in my pytest fixtures for `repository` and `reopen_repository`. I just pushed these fixtures - is this what you had in mind?